cmd/argocd/commands/app_test.go

```diff
@@ -963,3 +964,41 @@ func Test_unset_nothingToUnset(t *testing.T) {
 		})
 	}
 }
+
+func TestParseSelectedResources(t *testing.T) {
+	resources := []string{"v1alpha:Application:test", "v1alpha:Application:namespace/test"}
+	operationResources, err := parseSelectedResources(resources)
+	assert.NoError(t, err)
+	assert.Len(t, operationResources, 2)
+	assert.Equal(t, *operationResources[0], v1alpha1.SyncOperationResource{
+		Namespace: "",
+		Name:      "test",
+		Kind:      "Application",
+		Group:     "v1alpha",
+	})
+	assert.Equal(t, *operationResources[1], v1alpha1.SyncOperationResource{
+		Namespace: "namespace",
+		Name:      "test",
+		Kind:      "Application",
+		Group:     "v1alpha",
+	})
+}
+
+func TestParseSelectedResourcesIncorrect(t *testing.T) {
+	resources := []string{"v1alpha:test", "v1alpha:Application:namespace/test"}
+	_, err := parseSelectedResources(resources)
+	assert.EqualError(t, err, "Resource should have GROUP:KIND:NAME, but instead got: v1alpha:test")
+}
+
+func TestParseSelectedResourcesIncorrectNamespace(t *testing.T) {
+	resources := []string{"v1alpha:Application:namespace/test/unknown"}
+	_, err := parseSelectedResources(resources)
+	assert.EqualError(t, err, "Resource with namespace should have GROUP:KIND:NAMESPACE/NAME, but instead got: v1alpha:Application:namespace/test/unknown")
```

Comment:
The problem with tests asserting on error messages is that if someday someone changes the text, the test will fail. A better approach would be returning a proper error type and validating that during the tests. If having a dedicated that for that error is an overkill, we can check the string partially.